src/lib/components/Swap/SwapButton.tsx

```diff
@@ -171,7 +174,7 @@ export default function SwapButton({ disabled }: SwapButtonProps) {
       >
         <Trans>Review swap</Trans>
       </ActionButton>
-      {activeTrade && (
+      {activeTrade && onSupportedNetwork && (
```

Comment:
Does this mean that changing networks will close this dialog?
I think instead, changing networks should `setActiveTrade(undefined)`. Otherwise you'll still have an active trade for the wrong network (and switching between supported networks will show the old network's trade).

Reply:
This is actually already happening in the `useSyncSwapDefaults` hook. Any chain changes will clear fields to default values for the new chain.

Reply:
I'm still confused here - I think that any change to the chain should invalidate any active trade, which should be done as

```
useEffect(() => setActiveTrade(undefined), [chainId])
```

I don't think that `onSupportedNetwork` belongs in this file, as you wouldn't have an active trade if you aren't on a supported network (so it's a prerequisite to getting to this point).

Reply:
roughly, here's the logic flow:

`SwapButton.tsx`
```
const { trade, ... } = useSwapInfo()
...
useEffect(() => setActiveTrade(activeTrade => activeTrade && trade.trade), [trade])
```

`useSyncSwapDefaults`
```
const updateSwap = useUpdateAtom(swapAtom)

const setToDefaults = useCallback(() => updateSwap(DEFAULTS_FROM_INTEGRATOR))

useLayoutEffect(() => if (chainId && chainId !== previousChainId) setToDefaults(), [chainId, previousChainId, setToDefaults])
```

`useSwapInfo`
```
this is a lot of code, but it reads out swapAtom to build SwapInfo
any time swapAtom changes (like because the chain changes and triggers the useSyncSwapDefaults effect) it rebuilds SwapInfo. 

THIS is where we get a new trade entity. This is the trade.trade reference in the first code snippet. 
```

Reply:
chain change:
1. triggers useLayoutEffect
1. triggers setToDefaults
1. triggers updateSwap (resets values for swapAtom)
1. triggers rerun of useSwapInfo
1. clears calculated trade from `useBestTrade`
1. triggers rerun of the hook in SwapButton
2. calls `setActiveTrade((activeTrade) => truthyActiveTrade && undefined)`
3. activeTrade is now undefined